### PR TITLE
[Merged by Bors] - feat(CategoryTheory/SingleObj): characterise (co)limits of shape `SingleObj M` in types

### PR DESCRIPTION
**Pull request by chrisflav**
In the category of types: limits of shape `SingleObj M` are fixed points, colimits are quotients.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Shapes/SingleObj.lean [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    simp only [FunctorToTypes.map_comp_apply, types_comp_apply]
+    rfl
+
+section Limits
+
+variable (J : SingleObj M ⥤ Type u)
+
+/-- The equivalence between sections of `J : SingleObj M ⥤ Type u` and fixed points of the
+induced action on `J.obj (SingleObj.star M)`. -/
+def Types.sections.equivFixedPoints :
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/SingleObj.lean:53]: Does `@[simps]` generate useful simp-lemmas for this def? Idem for other defs...

**Review by jcommelin — COMMENTED** [review]
LGTM, modulo a question about `@[simps]`.

File: Mathlib/CategoryTheory/Limits/Shapes/SingleObj.lean (hunk 1 above)
Comment (chrisflav) [reply Mathlib/CategoryTheory/Limits/Shapes/SingleObj.lean:53]: For the ones about limits, yes. For the ones about colimits: Not sure, but the alternative simp lemma that I came up with is: `∀ x : J.obj (SingleObj.star G), Types.colimitEquivQuotient J (colimit.ι J (SingleObj.star G) x) = ⟦x⟧` is solved by simp if I add `@[simps!]` to `colimitEquivQuotient`, so I suppose simply adding `@[simps!]` is fine?

Comment (jcommelin) [reply Mathlib/CategoryTheory/Limits/Shapes/SingleObj.lean:54]: If you don't like the simp-lemmas that are auto-generated, you can of course manually prove better ones. But if you are reasonably happy with what `simps!` does, then I suggest we just move on. We can always come back and state manual simp-lemmas.

**Review by jcommelin — APPROVED** [review]
bors d+

**Pull request closed without merging**